gds/src/main/scala/edu/gemini/aspen/gds/Main.scala

```diff
@@ -0,0 +1,73 @@
+package edu.gemini.aspen.gds
+
+import cats.effect._
+import cats.effect.std._
+import cats.syntax.all._
+import fs2.Stream
+import scala.concurrent.duration._
+
+import edu.gemini.aspen.gds.configuration.KeywordConfiguration
+import edu.gemini.aspen.gds.keywords._
+import edu.gemini.aspen.gds.observations.{ ObservationManager, ObservationStateEvent }
+import edu.gemini.aspen.gds.seqexec.SeqexecServer
+import edu.gemini.aspen.giapi.status.StatusDatabaseService
+import edu.gemini.epics.EpicsReader
+
+object Main {
+  val purgeRate = 10.minutes // TODO: Add to configuration
+
+  val stream: Stream[IO, Unit] =
+    Stream.repeatEval(IO(println(java.time.LocalTime.now))).metered(30.second) //  .take(5)
+
+  def run(
+    keywordConfig:  KeywordConfiguration,
+    epicsReaderRef: Ref[IO, Option[EpicsReader]],
```

Comment:
Would it make sense to use a `Deferred[IO, EpicsReader]` to avoid the indirection of `Option`, or would it break with OSGi services coming and going?

Reply:
Right. Since a derferred can only be set once, we would be unable to "unset" it if the service goes away. I guess this isn't likely because of how gmp works, but I thought I should be conservative about it.